backend/routes.py

```diff
@@ -0,0 +1,145 @@
+def get_public_agents() -> list[dict]:
+    """
+    Get a list of all public agent types.
+    """
+    return [api_mocks.mock_agent_one_skill, api_mocks.mock_agent_two_skills]
+
+
+# 1.d.  User settings
+@pike.api.get("/user/{user_id}")
+def get_user_info(user_id: str) -> dict:
+    """
+    Provides full info about a user given their user_id.
+    """
+    return api_mocks.mock_user
+
+
+# 1.a.  get my agents
+@pike.api.get("/user/{user_id}/agents")
+def get_user_agents(user_id: u.UUID) -> list[dict]:
+    """
+    Get all agents the user has chosen to enable.
+    """
+    return api_mocks.mock_user["agents"]
+
+
+# 1.b.  get compact chat representations
+@pike.api.get("/user/{user_id}/chats")
+def get_user_chats_compact(user_id: str) -> list[dict]:
+    """
+    Gets a list of chat tags, providing enough information to render the associated
+    chats without messages.
+    """
+    return api_mocks.chat_without_messages, api_mocks.additional_chat_without_messages
+
+
+# 2.a
+@pike.api.get("/user/{user_id}/agent/{agent_id}/chats")
+def get_user_chats_by_agent(user_id: str, agent_id: u.UUID) -> list[dict]:
+    """
+    Return a list of all chats for a specific user which use a specific agent.
+    """
+    return api_mocks.chat_without_messages
+
+
+# 2.b (somwhat) get a chat's full history
+@pike.api.get("/user/{user_id}/chat/{chat_id}")
+def get_user_chat(user_id: str, chat_id: u.UUID) -> dict:
+    """
+    Provides the chat history for user {user_id} and thread {chat_id} in an
+    appropriate format for sending to the frontend.
+    """
+    return api_mocks.chat_with_messages
+
+
+# 2.h Get attachment information.  (Should be implicitly filtered by user as attachments are in user's chats)
+@pike.api.get("/attachment/{attachment_id}")
+def get_attachment(attachment_id: u.UUID) -> str:
+    """
+    Uses an attachment_id to request the data from a specific attachment from the backend.
+    """
+    return api_mocks.attachment_database.values()[0]
+
+
+@pike.api.get("/agent/{agent_id}")
+def get_agent(agent_id: u.UUID) -> dict:
+    """
+    Retrieve the information about a specific agent.
+    """
+    return api_mocks.mock_agent_two_skills
+
+
+## Create calls
+# 2.c Create a new chat
+@pike.api.post("/user/{user_id}/agent/{agent_id}/chat")
+def create_chat(user_id: str, agent_id: u.UUID) -> dict:
+    """
+    Generates a new, empty chat with a ChatID, attached to a specific user with a specific agent
+    employed within the chat.
+    """
+
+    return api_mocks.additional_chat_without_messages
+
+
+# c.ii.  add agent (Ignore agent preparedness flag for now)
+@pike.api.post("/user/{user_id}/agent/{agent_id}")
+def add_agent_to_user(user_id: str, agent_id: u.UUID) -> list[dict]:
+    """
+    Adds a new potential agent to the user's current agent list.
+    """
+    return api_mocks.mock_user["agents"].append(api_mocks.mock_agent_one_skill)
+
+
+# 2.d (and 2.c.ii)
+@pike.api.post("/user/{user_id}/chat/{chat_id}")
+def invoke_chat(
```

Comment:
let's call it `get_response`, it should be independent of what we use on the backend and invoke is a langchain specific interface.